addons/mail/static/src/utils/deferred/deferred.js

```diff
@@ -11,8 +11,8 @@ function makeDeferred() {
         resolve = res.bind(this);
         reject = rej.bind(this);
     });
-    prom.resolve = (...args) => resolve(...args);
-    prom.reject = (...args) => reject(...args);
+    prom.resolve = (...args) => { resolve(...args); return prom; };
+    prom.reject = (...args) => { reject(...args); return prom; };
```

Comment:
To make a resolved deferred, can just do `makeDeferred().resolve()`.
Useful to make such a deferred as default value for a parameter.
This actually matches the specs of deferred and promise to return a promise.

Reply:
Oh yeah... Forgot `Promise.resolve()` was a thing! Will use that instead.

Reply:
I think this should be reverted.

Native `Promise.resolve` is a class method whereas this implementation of reject is an instance method that looks like it can be called as the class method equivalent except it actually has an impact on its bound instance.

`Promise.resolve` either returns a new promise, or the existing promise unchanged. This one has a side effect.